src/fragments/lib/graphqlapi/js/setup/connect.mdx

```diff
@@ -1019,23 +977,170 @@ const { items: itemsPage2 } = res.data.listTodos;
+      id
+      title
+      description
+    }
+    nextToken
+  }
+}
+```
+
+ Now, to query the next 2 items from the API, you can pass this `nextToken` as the argument:
+
+```graphql
+query listTodos {
+  listTodos(limit: 2, nextToken: <your_token>) {
+    items {
+      id
+      title
+      description
+    }
+    nextToken
+  }
+}
+```
+
+When there are no other items left to be returned, the `nextToken` in the response will be set to null.
+
+##### Querying from a JavaScript application
+
+The `listTodos` query should have been created for you automatically by the CLI, but for reference purposes it should look something like this:
+
+```js
+const listTodos = `
+  query listTodos($limit: Int) {
+    listTodos(limit: $limit) {
+      items {
+        id
+        title
+        description
+      }
+      nextToken
+    }
+  }
+`
+```
+
+To pass in a `limit` in a query from a JavaScript application, you can use the following code by setting the limit as a variable:
+
+```js
+import { API } from 'aws-amplify';
+
+async function fetchTodos() {
+  const todoData = await API.graphql({
+    query: listTodos,
+    variables: {
+      limit: 2
+    }
+  })
+  console.log({ todoData })
+}
+```
+
+The data returned from the API request should look like this (with the items array containing however many items have been created):
+
+```graphql
+{
+  "data" {
+    "listTodos" {
+      "items": [{ id: "001", title: "Todo 1", description: "My first todo" }],
+      "nextToken": "<token-id>"
+    }
+  }
+}
+```
+
+To set the `nextToken` in a query from a JavaScript application, you can use the following code:
+
+```js
+import { API } from 'aws-amplify';
+
+async function fetchTodos() {
+  const todoData = await API.graphql({
+    query: listTodos,
+    variables: {
+      limit: 2,
+      nextToken: "<token-id>"
+    }
+  })
+  console.log({ todoData })
+}
+```
+
+</Accordion>
+
+<br/>
+
+<Callout>
+
+__Limitations:__
 - There is no API to get a total page count at this time. Note that scanning all items is a [potentially expensive operation](https://github.com/aws-amplify/amplify-js/issues/2901).
-- Sorting is [available in DataStore](https://docs.amplify.aws/lib/datastore/data-access/q/platform/js#predicates) but not in AppSync.
-- AppSync schemas do not follow the edges/nodes of the [Relay spec](https://relay.dev/docs/guides/graphql-server-specification) but are spiritually similar.
+- Sorting is [available in DataStore](https://docs.amplify.aws/lib/datastore/data-access/q/platform/js#predicates) but not in AWS AppSync.
+- AWS AppSync schemas do not follow the edges/nodes of the [Relay spec](https://relay.dev/docs/guides/graphql-server-specification) but are spiritually similar.
 - You [cannot query by `page` number](https://github.com/aws-amplify/amplify-cli/issues/5086), you have to query by `nextToken`.
-[comment]: # (RENE, as presented, these don't read as FAQs. This section needs to be reworked.)
+
+</Callout>
 
 ### Cancel read requests
```

Comment:
We should review where else this fragment is used and see if we want to change this and put it in the docs. It would mean we can make the paragraph more consistent and just replace code for other frameworks. This sample comes from https://github.com/aws-amplify/docs/blob/main/src/fragments/lib/graphqlapi/js/cancel-request.mdx